HH-452 | Clean up test output

karisal-anders opened the pull request:
## Description

Clean up hobbies-helsinki app's tests' output.

Only remaining errors/warnings in hobbies-helsinki app's tests' output are:
1. `Warning: Each child in a list should have a unique "key" prop.`
   - Check the render method of `CarouselSlider`. See https://reactjs.org/link/warning-keys for more information.
2. `Warning: validateDOMNesting(...): <a> cannot appear as a descendant of <a>.`

Fixing these errors/warning may need fixes to [RHHC](https://github.com/City-of-Helsinki/react-helsinki-headless-cms/)

This PR is **quite similar** to **PR #870**

## Related

[HH-452](https://helsinkisolutionoffice.atlassian.net/browse/HH-452)

[HH-452]: https://helsinkisolutionoffice.atlassian.net/browse/HH-452?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

karisal-anders committed:
feat(tests): enable readable apollo client errors in hobbies-helsinki

Previously tests showed errors like:
```
An error occurred! For more details, see the full error text at
https://go.apollo.dev/c/err#%7B...
```
where the link went on for many rows.

This enables the apollo client errors messages to be directly readable
rather than having to click on a link and go to a webpage where one can
see the error message.

refs HH-452

karisal-anders committed:
fix(tests): hide sourcemap missing warnings in hobbies-helsinki tests

This declutters output of apps/hobbies-helsinki tests.

refs HH-452

karisal-anders committed:
fix(tests): cleanup hobbies AppConfig tests & their output

Previously running hobbies AppConfig.test.tsx tests there was output:
```
stdout | src/domain/app/__tests__/AppConfig.test.tsx > provides number
config 'defaultRevalidate'
Could not parse env value SyntaxError: Unexpected token 'S', "Some
string" is not valid JSON
```

This removes that output from running the tests & instead tests for it.

Also remove all @ts-ignore from AppConfig.test.tsx & fix the related
typings.

refs HH-452

karisal-anders committed:
fix(tests): simplify EventPageContainer test about event w/o keywords

refs HH-452

karisal-anders committed:
fix(tests): use unique IDs in hobbies EventPageContainer tests

This change removes Apollo client cache warning when running hobbies
EventPageContainer tests.

refs HH-452

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr871.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr871.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr871.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr871.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr871.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr871.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr871.dev.hel.ninja :laughing::tada::tada::tada:

karisal-anders commented:
Finally everything went through in CI/CD pipeline as I deleted some DeploymentConfigs from the dev environment. I think the infra side is running to its limits with our current setup and cleaning up of unnecessary/unused/already merged PR environments is not working or if it is working it is at least not working fast enough.

karisal-anders commented:
So, to be clear: this PR is now ready for code review.

karisal-anders committed:
fix(tests): fix hobbies Search "for accessibility violations" test

Also remove 50s timeout from test case as unnecessary.

Previously running hobbies Search.test.tsx tests there was ~output:
```
$ yarn test src/domain/search/eventSearch/advancedSearch/__tests__/
Search.test.tsx --watch

 DEV  v3.2.4 apps/hobbies-helsinki

stderr | src/domain/search/eventSearch/advancedSearch/__tests__/
Search.test.tsx > for accessibility violations

Error: Not implemented: HTMLCanvasElement.prototype.getContext (without
installing the canvas npm package)
...

Warning: An update to PlaceSelector inside a test was not wrapped in
act(...).

When testing, code that causes React state updates should be wrapped
into act(...):

act(() => {
  /* fire events that update state */
});
/* assert on the output */

This ensures that you're testing the behavior the user would see in the
browser. Learn more at https://reactjs.org/link/wrap-tests-with-act
    at PlaceSelector (packages\components\src\components\domain\place\
    placeSelector\PlaceSelector.tsx:63:3)
    ...
```

refs HH-452

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr871.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr871.dev.hel.ninja :laughing::tada::tada::tada:
